### PR TITLE
Add Support for RN 0.71

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Reflect changes from RN 0.71 in the sourcemaps upload gradle task.

### Motivation

Add support for RN 0.71

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Review by mariusc83 — APPROVED** [review]
LGTM

File: packages/core/datadog-sourcemaps.gradle [hunk 1]
```diff
@@ -33,7 +33,8 @@ afterEvaluate {
         def targetName = variant.name.capitalize()
         def targetPath = variant.dirName
 
-        def bundleTask = tasks.findByName("bundle${targetName}JsAndAssets")
+        // Could be retro-compatible: check
```
Comment (louiszawadzki) [inline packages/core/datadog-sourcemaps.gradle:36]: Tested: not retro-compatible

File: packages/core/datadog-sourcemaps.gradle [hunk 2]
```diff
@@ -47,9 +48,11 @@ afterEvaluate {
         def serviceName = getServiceName(variant)
         logger.info("Service name used for the upload of variant=${targetName} is ${serviceName}.")
 
-        def bundleAssetName = reactConfig.bundleAssetName ?: "index.android.bundle"
+        // Could be retro-compatible: check
```
Comment (louiszawadzki) [inline packages/core/datadog-sourcemaps.gradle:51]: Tested: not retro-compatible

File: packages/core/datadog-sourcemaps.gradle [hunk 3]
```diff
@@ -47,9 +48,11 @@ afterEvaluate {
...
+        def bundleAssetName = reactConfig.bundleAssetName.get() ?: "index.android.bundle"
 
-        def jsBundleDir = file("$buildDir/generated/assets/react/${targetPath}")
+        // Probably a bug in RN
+        def jsBundleDir = file("$buildDir/ASSETS/createBundle${targetPath}JsAndAssets")
```
Comment (louiszawadzki) [inline packages/core/datadog-sourcemaps.gradle:55]: Check this still works on 0.71.0-rc5

**Review by 0xnm — APPROVED** [review]

File: packages/core/datadog-sourcemaps.gradle [hunk 4]
```diff
@@ -98,6 +98,63 @@ afterEvaluate {
...
+ */
+private def getReactConfig(File buildDir, String targetName) {
+    def reactConfig = [:]
+
+    if (project.extensions.findByName("react")) {
+        // From RN 0.71
+        def bundleAssetName = project.extensions.react.bundleAssetName.get() ?: "index.android.bundle"
+        reactConfig['bundleTaskName'] = "createBundle${targetName}JsAndAssets"
+        reactConfig['bundleFileResolver'] = getBundleFileResolver([
+            "$buildDir/ASSETS/createBundle${targetName}JsAndAssets", // Android Gradle Plugin 7.3
```
[the remark this answers is not in the document]
Comment (louiszawadzki) [reply packages/core/datadog-sourcemaps.gradle:133]: The generated directory is `ASSETS`, not `assets` - it is a bug in AGP 7.3 that is fixed in 7.4. So I believe the change won't work with Linux?
I think searching recursively might be more complex to implement and to debug as well.

[the remark this answers is not in the document]
Comment (0xnm) [reply packages/core/datadog-sourcemaps.gradle:133]: good as it is then :) I was thinking `ASSETS` is a wrong one.

[the remark this answers is not in the document]
Comment (louiszawadzki) [reply packages/core/datadog-sourcemaps.gradle:133]: linking the AGP fix for future reference: https://android.googlesource.com/platform/tools/base/+/374b58dfff28dbaccbb2815db66c412b0bccf013%5E%21/build-system/gradle-core/src/main/java/com/android/build/api/variant/impl/SourceDirectoriesImpl.kt

**Pull request merged**